extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/CommonPanacheQueryImpl.java

```diff
@@ -198,9 +204,24 @@ private String countQuery() {
         Matcher selectMatcher = SELECT_PATTERN.matcher(query);
         String countQuery;
         if (selectMatcher.matches()) {
-            countQuery = "SELECT COUNT(" + selectMatcher.group(1) + ") " + selectMatcher.group(2);
-        } else {
+            // this one cannot be null
+            String firstSelection = selectMatcher.group(1).trim();
+            if (firstSelection.toLowerCase().startsWith("distinct ")) {
+                // this one can be null
+                String secondSelection = selectMatcher.group(2);
+                // we can only count distinct single columns
+                if (secondSelection != null && !secondSelection.trim().isEmpty()) {
+                    throw new PanacheQueryException("Count query not supported for select query: " + query);
+                }
+                countQuery = "SELECT COUNT(" + firstSelection + ") " + selectMatcher.group(3);
+            } else {
+                // it's not distinct, forget the column list
+                countQuery = "SELECT COUNT(*) " + selectMatcher.group(3);
+            }
+        } else if (FROM_PATTERN.matcher(query).matches()) {
             countQuery = "SELECT COUNT(*) " + query;
+        } else {
+            throw new PanacheQueryException("Count query not supported for select query: " + query);
```

Comment:
I generally think it's a good idea to enclose this type of output in quotes, but I'll leave it up to you :)